sdk/core/Azure.Core.Experimental/src/Primitives/BinaryDataExtensions.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Experimental.Primitives
+{
+    /// <summary>
+    /// A set of extension methods that can be used to access the payload
+    /// of <see cref="BinaryData"/>.
+    /// </summary>
+    public static class BinaryDataExtensions
+    {
+        /// <summary>
+        /// Converts the BinaryData to a string using UTF-8.
+        /// </summary>
+        /// <param name="data">The BinaryData instance.</param>
+        /// <returns>The string representation of the data.</returns>
+        public static string AsString(this BinaryData data) =>
+           data.AsString(Encoding.UTF8);
+
+        /// <summary>
+        /// Converts the BinaryData to a string using the specified
+        /// encoding.
+        /// </summary>
+        /// <param name="data">The BinaryData instance.</param>
+        /// <param name="encoding">The encoding to use when decoding
+        /// the bytes.</param>
+        /// <returns>The string representation of the data.</returns>
+        public static string AsString(
+            this BinaryData data,
+            Encoding encoding)
+        {
+            Argument.AssertNotNull(encoding, nameof(encoding));
+            return encoding.GetString(data.Memory.ToArray());
+        }
+
+        /// <summary>
+        /// Converts the BinaryData to bytes.
+        /// </summary>
+        /// <param name="data">The BinaryData instance.</param>
+        /// <returns>The data as bytes.</returns>
+        public static ReadOnlyMemory<byte> AsBytes(
+            this BinaryData data) =>
+            data.Memory;
+
+        /// <summary>
+        /// Converts the BinaryData to a stream.
+        /// </summary>
+        /// <param name="data">The BinaryData instance.</param>
+        /// <returns>A stream representing the data.</returns>
+        public static Stream AsStream(this BinaryData data) =>
+            new MemoryStream(data.Memory.ToArray());
```

Comment:
I have a crazy suggestion for you.  Don't do it now, but maybe we create a private stream class that wraps a MemoryStream so customers can't cast to MemoryStream and depend on that behavior?  It would give us the freedom to grow here down the road.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/12337